### PR TITLE
Chore: Adding Black + Flake8 Linters

### PR DESCRIPTION
**Pull request by guanjieshen**
**_Why:_**
Standardize linters and code quality check for Python code within the repo. 

Leveraging Black and Flake8 (however this is up for discussion on if it's something the team would want to adopt.)

**_What:_**
- Ran `black python` on the Python directory to format all code according to Black style guide.
- Added 'black` linter check into the github actions workflow
- Added 'flake8` linter check into the github actions workflow

To run flake8 locally use `flake8 python --config python/.flake8`.
To run black locally use `black python`

**_Additional Considerations:_**

Right now there's are still a lot of flake8 rule violations. We can choose which ones we want to address or ignore. Up for suggestions :) For now, to skip flake8 checks for a specific row we can just add the following:

```python
# flake8: noqa: F821
dbutils.fs.ls("/")
```

To skip a certain rule globally, we can add it to the `extend-ignore` list in the `.flake8` folder.

I am finding that Black does not do a good job with long arrays and forcibly wraps them; thereby making it difficult to read. This issue does not appear like it will be address by the Black team moving forward https://github.com/psf/black/issues/1281. 

For example, Black turns the following:
```python
        expected_data = [
            ["S1", "2020-08-01 00:00:10", 349.21, "2020-08-01 00:00:01", 345.11, 351.12],
            ["S1", "2020-08-01 00:01:12", 351.32, "2020-08-01 00:01:05", 348.10, 353.13],
```
into
```python
    expected_data = [
        [
            "S1",
            "2020-08-01 00:00:10",
            349.21,
            "2020-08-01 00:00:01",
            345.11,
            351.12,
        ],
        [
            "S1",
            "2020-08-01 00:01:12",
            351.32,
            "2020-08-01 00:01:05",
            348.10,
            353.13,
        ],
```
For readability sake, if there's an array were we want to ignore black formatting rules we can just skip the linter check e.g.

```python
        # fmt: off
        expected_data = [
                ["S1", "2020-08-01 00:00:10", 349.21, "2020-08-01 00:00:01", 345.11, 351.12],
                ["S1", "2020-08-01 00:01:12", 351.32, "2020-08-01 00:01:05", 348.10, 353.13],
                ["S1", "2020-09-01 00:02:10", 361.1, "2020-09-01 00:02:01", 358.93, 365.12],
                ["S1", "2020-09-01 00:19:12", 362.1, "2020-09-01 00:15:01", 359.21, 365.31]]
        # fmt: on
```

### CODE REVIEW

**Comment by guanjieshen** [comment]
Some alerts from flake8 around unused variables:

```
./python/tests/tsdf_tests.py:194:9: F841 local variable 'dfRight' is assigned to but never used
./python/tests/tsdf_tests.py:195:9: F841 local variable 'dfExpected' is assigned to but never used
./python/tests/tsdf_tests.py:1214:9: F841 local variable 'expectedSchema' is assigned to but never used
./python/tests/tsdf_tests.py:1261:9: F841 local variable 'interpolated' is assigned to but never used
./python/tests/tsdf_tests.py:1388:9: F841 local variable 'dfExpected' is assigned to but never used
```

**Comment by tnixon** [comment]
@guanjieshen looks like you need to resolve merge conflicts

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,6 +7,30 @@ on:
...
+      - name: Set up Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+      - name: flake8 Lint
+        uses: py-actions/flake8@v2
+        with:
+          args: "--config python/.flake8"
+          path: "./python"
   run:
```
Comment (tnixon) [inline .github/workflows/test.yml:34]: Now that we have multiple jobs in this workflow, let's give this one a more descriptive name than just "run". Maybe "test" and `name: Build and Test Module`

**Review by tnixon — CHANGES_REQUESTED** [review]
@guanjieshen - there seem to be a lot of changes that are just cosmetic formatting of code. It makes it hard to distinguish the important changes for this functionality from general code cleanup. Could you please submit just the necessary changes, and leave out any formatting. 

I'm not against code cleanup when it aids in readability, but those should be submitted as their own PR and not crammed into an unrelated change.

**Comment by tnixon** [comment]
... or are these formatting changes in order to get Flake8 to pass?

**Comment by guanjieshen** [comment]
@tnixon going to close this PR for now and start a new one to see if we can better layer in the linter changes rather than have it in a giant PR.

**Pull request closed without merging**